nominatim/tools/special_phrases.py

```diff
@@ -120,9 +120,10 @@ def _check_sanity(lang, phrase_class, phrase_type, pattern):
     """
     try:
         if len(pattern.findall(phrase_class)) < 1 or len(pattern.findall(phrase_type)) < 1:
-            sys.exit()
-    except SystemExit:
-        LOG.error("Bad class/type for language %s: %s=%s", lang, phrase_class, phrase_type)
+            raise UsageError("Bad class/type for language {}: {}={}".format(
+                lang, phrase_class, phrase_type))
+    except UsageError:
+        LOG.error(UsageError)
```

Comment:
You can also simply put the `LOG.error()` before the raise and get rid of the whole try/except/raise pattern.